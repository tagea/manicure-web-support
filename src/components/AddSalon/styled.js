import styled from 'styled-components';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import GridList from '@material-ui/core/GridList';
import GridListTileBar from '@material-ui/core/GridListTileBar';
import Button from '@material-ui/core/Button';

export const FormBlockStyled = styled(Grid)`
  padding-top: 10px;
`;

export const DayStyled = styled(Typography)`
  text-transform: capitalize;
  text-align: right;
  &&{
    margin-right: 10px;
    line-height: 46px;
  }
`;

export const BlockPhotosStyled = styled(Grid)`
  overflow: hidden;
  height: 190px;
`;

export const GridListStyled = styled(GridList)`
  &&{
    flex-wrap: nowrap;
  }
  transform: translateZ(0);
`;

export const ButtonAddStyled = styled(Grid)`
  text-align: center;
  position: relative;
  top: calc(90px - 40px / 2);
`;

export const GridListTileBarStyled = styled(GridListTileBar)`
  div{
    flex-grow: 1;
  }
`;

export const InputFileStyled = styled('input')`
  display: none;
`;

export const AddPhotoStyled = styled(Button)`
  &&{
    display: inline-block;
    height: 40px;
    border-radius: 50%;
    min-width: 40px;
    width: 40px;
    padding: 9px;
  }
`;
