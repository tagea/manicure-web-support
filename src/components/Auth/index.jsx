import React from 'react';
import PropTypes from 'prop-types';

import Grid from '@material-ui/core/Grid';
import { Logo } from '../../ui/Logo';
import { Input } from '../../ui/Input';
import { Button } from '../../ui/Button';
import { BackgroundStyled } from '../../globalStyled';
import { PaperStyled } from './styled';

const Auth = (props) => {
  const { error, changeInput, pressEnter, handleSubmit } = props;
  return (
    <BackgroundStyled container>
      <Grid container direction="column" justify="center" alignItems="center">
        <Logo />
        <PaperStyled container direction="column" justify="center" alignItems="center">
          <Input
            id="username-support"
            label="Username"
            type="text"
            name="username"
            autoComplete="username"
            size="large"
            error={error}
            onChange={changeInput}
          />
          <Input
            id="password-support"
            label="Password"
            type="password"
            name="password"
            autoComplete="current-password"
            size="large"
            error={error}
            onChange={changeInput}
            onKeyDown={pressEnter}
          />
          <Button variant="contained" size="large" color="secondary" text="Sign in" handleClick={handleSubmit} />
        </PaperStyled>
      </Grid>
    </BackgroundStyled>
  );
};

Auth.propTypes = {
  error: PropTypes.bool,
  handleSubmit: PropTypes.func,
  changeInput: PropTypes.func,
  pressEnter: PropTypes.func
};
Auth.defaultProps = {
  error: false,
  handleSubmit: () => false,
  changeInput: () => false,
  pressEnter: () => false
};

export { Auth };
