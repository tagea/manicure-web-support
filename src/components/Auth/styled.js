import styled from 'styled-components';
import Grid from '@material-ui/core/Grid';
import { theme } from '../../ui/Themes';

const { palette } = theme;

export const PaperStyled = styled(Grid)`
  &&{
    width: 324px;
  }
  border-radius: 4px;
  background-color: ${palette.surface.main};
  padding: 42px 27px;
`;
