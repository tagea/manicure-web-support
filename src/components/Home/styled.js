import styled from 'styled-components';
import { Typography } from '@material-ui/core';

export const TopHeaderStyled = styled(Typography)`
  flex-grow: 1;
`;
