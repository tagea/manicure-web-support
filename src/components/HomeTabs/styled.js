import styled from 'styled-components';
import Grid from '@material-ui/core/Grid';
import AppBar from '@material-ui/core/AppBar';
import Tab from '@material-ui/core/Tab';
import { theme } from '../../ui/Themes';

const { palette } = theme;

export const ContainerStyled = styled(Grid)`
  flex: auto;
  overflow: hidden;
`;

export const TabContainerStyled = styled(Grid)`
  height: 100%;
  margin-top: 30px;
`;

export const AppBarStyled = styled(AppBar)`
  &&{
    box-shadow: none;
    border-radius: 3px;
    position: relative;
    >button{
      top: 4px;
    }
  }
`;

export const TabStyled = styled(Tab)`
  &&{
    font-size: 1.2em;
    padding: 6px 0;
    color: ${palette.text.main};
  }
`;

export const LogOutButtonStyled = styled(Grid)`
  position: absolute;
  right: 20px;
  top: 4px;
`;
