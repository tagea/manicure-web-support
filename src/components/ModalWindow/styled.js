import styled from 'styled-components';
import Typography from '@material-ui/core/Typography';
import DialogTitle from '@material-ui/core/DialogTitle';
import { theme } from '../../ui/Themes';

const { palette } = theme;

export const GroupButtonStyled = styled(Typography)`
  button{
    margin: 0 4px;
  }
`;

export const DialogTitleStyled = styled(DialogTitle)`
  border-bottom: 1px solid ${palette.border.light};
  &&{
    padding-bottom: 26px;
    margin-bottom: 30px;
  }
`;
