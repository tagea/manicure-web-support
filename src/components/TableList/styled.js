import styled from 'styled-components';
import Paper from '@material-ui/core/Paper';
import TableCell from '@material-ui/core/TableCell';
import Grid from '@material-ui/core/Grid';

import { Table } from 'react-virtualized';
import { theme } from '../../ui/Themes';

const { palette } = theme;

export const PaperStyled = styled(Paper)`
  &&{
    background-color: ${palette.surface.light};
    border-radius: 3px;
    box-shadow: none;
    height: 100%;
  }
`;

export const TableStyled = styled(Table)`
  font-family: 'Roboto', sans-serif;
  font-weight: 300;
  box-sizing: border-box;
  div:focus{
      outline: none;
  }
  .rowTable{
    display: flex;
    padding: 30px 25px;
    box-sizing: border-box;
    border-bottom: 1px solid ${palette.border.light};
    border-top: 1px solid transparent;
    transition: all 0.2s ease-in;
    :hover{
      background-color: ${palette.surface.transparent};
      border-bottom: 1px solid ${palette.border.main};
      border-top: 1px solid ${palette.primary.main};
    }
  }
  .ReactVirtualized__Table__headerRow:hover{
    background-color: transparent;
    border-bottom: 1px solid ${palette.border.light};
    border-top: 1px solid transparent;
    
  }
  .headerStyle{
    padding-top: 0;
    padding-bottom: 0;
    line-height: 20px;
    color: ${palette.text.transparent};
  }
`;

export const TableCellStyled = styled(TableCell)`
  &&{
    font-size: 1.1em;
    border-bottom: none;
    padding: 0px;
  }
`;

export const TableHeader = styled(Grid)` 
  &&{
    display:flex;
    justify-content: space-between;
    margin: 0 20px;
    padding-top: 25px;
  }
`;
