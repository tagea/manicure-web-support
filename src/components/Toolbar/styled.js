import styled from 'styled-components';
import AppBar from '@material-ui/core/AppBar';

export const AppBarStyled = styled(AppBar)` 
  &&{
      background-color: transparent;   
      box-shadow: none;                  
      align-items: flex-end;
      justify-content: center;
      padding-top: 10px;
  }
`;
