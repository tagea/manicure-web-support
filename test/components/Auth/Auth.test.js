import React from 'react';
import { shallow } from 'enzyme';
import { Auth } from '../../../src/components/Auth';
import { HeaderStyled, BackgroundStyled } from '../../../src/globalStyled';
import { PaperStyled } from '../../../src/components/Auth/styled';

/* create Auth snapshot */
it('render Auth component', () => {
  const AuthComponent = shallow(<Auth />);
  expect(AuthComponent).toMatchSnapshot();
});

/* test for BackgroundStyled prop */
it('Auth as wrapper BackgroundStyled should contain value "container"', () => {
  const AuthComponent = shallow(<Auth />);
  expect(AuthComponent.find(BackgroundStyled).prop('container')).toBeTruthy();
});

/* test for PaperStyled container propery */
it('PaperStyled should contain "container" property', () => {
  const AuthComponent = shallow(<Auth />);
  expect(AuthComponent.find(PaperStyled).prop('container')).toBeTruthy();
});

/* test for PaperStyled direction propery */
it('PaperStyled should contain "direction" property width "column" value', () => {
  const AuthComponent = shallow(<Auth />);
  expect(AuthComponent.find(PaperStyled).prop('direction')).toEqual('column');
});

/* test for Input type */
it('Input type should be text and password', () => {
  const AuthComponent = shallow(<Auth />);
  const input = AuthComponent.find('Input').map(node => node.prop('type'));
  expect(input).toEqual(['text', 'password']);
});

/* test for Input error */
it('Input error should be password type', () => {
  const AuthComponent = shallow(<Auth />);
  const input = AuthComponent.find('Input').map(node => node.prop('error'));
  expect(input).toEqual([false, false]);
});

/* test Input functions */
it('Input username should have onChange function', () => {
  const AuthComponent = shallow(<Auth />);
  const input = AuthComponent.find('Input').first().prop('onChange');
  expect(input).toBeInstanceOf(Function);
});
it('Input password should have onChange function', () => {
  const AuthComponent = shallow(<Auth />);
  const input = AuthComponent.find('Input').last().prop('onChange');
  expect(input).toBeInstanceOf(Function);
});

/* test Button text */
it('Button should contain value for "text" property', () => {
  const AuthComponent = shallow(<Auth />);
  expect(AuthComponent.find('Button').prop('text')).not.toHaveLength(1);
});

/* test Button function */
it('Button should have onCLick function', () => {
  const AuthComponent = shallow(<Auth />);
  expect(AuthComponent.find('Button').prop('handleClick')).toBeInstanceOf(Function);
});
