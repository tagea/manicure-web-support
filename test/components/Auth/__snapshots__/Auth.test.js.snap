// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Auth component 1`] = `
<globalStyled__BackgroundStyled
  container={true}
>
  <WithStyles(Grid)
    alignItems="center"
    container={true}
    direction="column"
    justify="center"
  >
    <Logo />
    <styled__PaperStyled
      alignItems="center"
      container={true}
      direction="column"
      justify="center"
    >
      <Input
        autoComplete="username"
        error={false}
        id="username-support"
        label="Username"
        name="username"
        onChange={[Function]}
        type="text"
      />
      <Input
        autoComplete="current-password"
        error={false}
        id="password-support"
        label="Password"
        name="password"
        onChange={[Function]}
        onKeyDown={[Function]}
        type="password"
      />
      <Button
        color="secondary"
        handleClick={[Function]}
        size="large"
        text="Sign in"
      />
    </styled__PaperStyled>
  </WithStyles(Grid)>
</globalStyled__BackgroundStyled>
`;
